motr/client_init.c

```diff
@@ -1687,7 +1692,15 @@ int m0_client_init(struct m0_client **m0c_p,
 			size = conf->mc_addb_size;
 			M0_LOG(M0_DEBUG, "ADDB size = %" PRIu64 "", size);
 		}
-		sprintf(buf, "linuxstob:./addb_%d", (int)m0_pid());
+#ifndef __KERNEL__
+		conf->mc_addb_stob_location = getenv("M0_CLIENT_ADDB_DIR");
```

Comment:
local variable can be used instead of configuration structure conf->mc_addb_stob_location

Reply:
I tried based on your request, It doesn't meet our requirement like below:

```
[root@ssc-vm-rhev4-2700 ac2f9656e38a92990522025e1f3ccaef]# tree
.
├── addb_files-0x7200000000000001:0x7
├── motr_trace_files
│   ├── m0trace.10.2022-08-19-07:55:46
│   └── m0trace.398.2022-08-19-07:54:55
├── radosgw-admin.log
├── rgw-1.log
├── rgw_debug
│   ├── addb_10
│   │   ├── id
│   │   └── o
│   │       └── 100000000000000:2
│   └── addb_398
│       ├── id
│       └── o
│           └── 100000000000000:2
├── rgw_setup.log
└── rgw_startup.log
7 directories, 10 files
[root@ssc-vm-rhev4-2700 ac2f9656e38a92990522025e1f3ccaef]#
```


If we use variables in conf structure, then our result will be like below: 

```
[root@ssc-vm-rhev4-2700 ac2f9656e38a92990522025e1f3ccaef]# ls
addb_files-0x7200000000000001:0x7  motr_trace_files  radosgw-admin.log  rgw-1.log  rgw_debug  rgw_setup.log  rgw_startup.log
[root@ssc-vm-rhev4-2700 ac2f9656e38a92990522025e1f3ccaef]# tree
.
├── addb_files-0x7200000000000001:0x7
│   ├── addb_11
│   │   ├── id
│   │   └── o
│   │       └── 100000000000000:2
│   └── addb_398
│       ├── id
│       └── o
│           └── 100000000000000:2
├── motr_trace_files
│   ├── m0trace.11.2022-08-18-12:27:18
│   └── m0trace.398.2022-08-18-12:26:59
├── radosgw-admin.log
├── rgw-1.log
├── rgw_debug
├── rgw_setup.log
└── rgw_startup.log
7 directories, 10 files
[root@ssc-vm-rhev4-2700 ac2f9656e38a92990522025e1f3ccaef]#
```